ui/src/context/terminalContext.ts

```diff
@@ -195,10 +208,13 @@ export const createTerminalContext = (): TerminalContext => {
   return {
     eventBus,
     lunaCommunicator,
+
+    cleanup,
+    initialize,
     sendLunaEvent,
+    sendMittEvent,
+    onMittEvent,
     initializeLunaListeners,
-    initialize,
-    cleanup,
   };
 };
 
```

Comment:
There is one issue with this code that needs to be addressed. The `cleanup` property seems redundant since it's defined after both the `initialize` and `initializeLunaListener`s properties (which would also call `cleanup`). The correct structure should look something like:

```typescript
const initialize = () => {};
const cleanup = () => {};

// The remaining properties can remain unchanged
```

This ensures that before calling `initializeLunaListener`, there is at least one instance of `initialize`. Otherwise, we may accidentally enter an infinite loop where we try to unsubscribe from events multiple times.

After addressing this single issue, please let me know if you want to discuss further improvements or optimizations within the context of these changes.